### PR TITLE
added test cases for Book and Category entities on each layer

### PR DESCRIPTION
**Pull request by knyazkyylubomir**
test coverage result

### CODE REVIEW

File: src/test/java/org/project/name/online/book/store/controller/BookControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class BookControllerTest {
```
Comment (Ivan95kos) [inline src/test/java/org/project/name/online/book/store/controller/BookControllerTest.java]:
```suggestion
class BookControllerIntegrationTest {
```

File: src/test/java/org/project/name/online/book/store/controller/CategoryControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+
```
Comment (Ivan95kos) [inline src/test/java/org/project/name/online/book/store/controller/CategoryControllerTest.java]: line
```suggestion
```

File: src/test/java/org/project/name/online/book/store/controller/CategoryControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+class CategoryControllerTest {
```
Comment (Ivan95kos) [inline src/test/java/org/project/name/online/book/store/controller/CategoryControllerTest.java]:
```suggestion
class CategoryControllerIntegrationTest {
```

File: src/test/java/org/project/name/online/book/store/repository/BookRepositoryTest.java [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.test.context.jdbc.Sql;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
+
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+class BookRepositoryTest {
```
Comment (Ivan95kos) [inline src/test/java/org/project/name/online/book/store/repository/BookRepositoryTest.java]:
```suggestion
class BookRepositoryIntegrationTest {
```

File: src/test/java/org/project/name/online/book/store/controller/BookControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+        requestDto.setCoverImage("Image");
+        requestDto.setCategoryIds(List.of(1L));
+        BookDto expected = new BookDto();
+        expected.setTitle(requestDto.getTitle());
+        expected.setAuthor(requestDto.getAuthor());
+        expected.setIsbn(requestDto.getIsbn());
+        expected.setPrice(requestDto.getPrice());
+        expected.setDescription(requestDto.getDescription());
+        expected.setCoverImage(requestDto.getCoverImage());
+        expected.setCategoryIds(List.of("Test name1"));
```
Comment (Ivan95kos) [inline src/test/java/org/project/name/online/book/store/controller/BookControllerTest.java]: it is difficult to read such tests
It is better to transfer all preparation to a private method with the appropriate name
fix it everywhere

**Review by Ivan95kos — CHANGES_REQUESTED** [review]
Good job overall!

**Review by kozhukhovsky — APPROVED** [review]

**Pull request merged**